scripts/esb_sniffer/Sniffer.py

```diff
@@ -0,0 +1,249 @@
+            return -1
+
+        if len(ret) == 4:
+            length = int.from_bytes(ret, byteorder='big', signed=False)
+        else:
+            self.logger.error("Failed to get packet length from sniffer, using default %d", self.packet_len)
+            return -1
+
+        if length != 0:
+            self.packet_len = length
+
+        return 0
+
+    def connect(self) -> int:
+        '''Connect to the DK.'''
+        timeout = 5
+
+        if self.jlink is not None:
+            return 0
+
+        try:
+            self.jlink = API()
+            self.jlink.open()
+            self.jlink.connect_to_emu_without_snr(jlink_speed_khz=self.swd_freq_khz)
+
+            if self.jlink.read_device_family() == "NRF53":
+                self.jlink.select_coprocessor(CoProcessor.CP_NETWORK)
+
+            self.jlink.sys_reset()
+            self.jlink.go()
+            self.jlink.rtt_start()
+        except APIError as err:
+            self.logger.error("Failed to connect to device: err = %d", err.err_code)
+            self.jlink = None
+            return -1
+
+        while (not self.jlink.rtt_is_control_block_found() and timeout > 0):
+            sleep(1)
+            timeout -= 1
+
+        if timeout == 0:
+            self.logger.error("Failed to find rtt control block")
+            self.jlink = None
+            return -1
+
+
+        self.__get_packet_length()
+        self.logger.info("Connected to device")
+        return 0
+
+    def disconnect(self):
+        '''Disconnect from the DK.'''
+        try:
+            self.jlink.close()
+        except APIError:
+            self.logger.error("JLink connection lost")
+
+        self.jlink = None
+        self.logger.info("Disconnected from device")
+
+    def __gen_pcap_packets(self, buff: bytearray) -> list:
+        '''Parse data from DK into pcap formatted packets.'''
+        if len(buff) < self.packet_len:
+            return b''
+
+        packets = []
+
+        # Since rtt_read(self.packet_len) can read fragment of the packet,
+        # ensure that we will parse it correctly
+        if self.remaing_data is not None:
+            buff = self.remaing_data + buff
+            self.remaing_data = None
+
+        if len(buff) % self.packet_len != 0:
+            start = int(len(buff)/self.packet_len) * self.packet_len
+            self.remaing_data = buff[start:]
+
+        for i in range(0, int(len(buff)/self.packet_len)):
+            pkt = buff[i*self.packet_len:(i+1)*self.packet_len]
+            if pkt != b'':
+                timestamp_ms = int.from_bytes(pkt[0:4], byteorder='big', signed=False)
+                timestamp_us = int.from_bytes(pkt[4:8], byteorder='big', signed=False)
+
+                sec = int(timestamp_ms / 1000)
+                usec = int(timestamp_us+((timestamp_ms % 1000) * 1e3))
+                data_length = pkt[8]
+                pkt_len = 5 + data_length
+
+                pkt_header = struct.pack('@IIII', sec, usec, pkt_len, pkt_len)
```

Comment:
The packet header is packed as '@IIII'. For cross-platform compatibility, consider using a fixed endianness. I suggest '<' for little-endian rather than native ('@').